### ISSUE TITLE
[Semantic API] Correct type symbol not returned when a constant is used as a variable type

### ISSUE THREAD

**Issue by pubudu91** [ballerina-platform/ballerina-lang#27190]
When a constant is used as a type when defining a variable, the type of the var symbol is set as the resolved type, not the constant symbol. For example,
```ballerina
public const RSA = "RSA";

function testExprs()  {
    RSA rsa = "RSA";
}
```
In the above case, when you check the type of the `rsa` var symbol, it should give the constant symbol. But currently, it gives the resolved singleton type. Since a constant reference is a singleton type, the above should've returned the constant symbol.

**Comment by pubudu91** [comment]
Sort of related to #26284

**Comment by pubudu91** [comment]
This seems to get fixed with Chiran's PR. Better test this further at the Semantic API level as well.
<img width="556" alt="Screenshot 2021-10-30 at 00 45 55" src="https://user-images.githubusercontent.com/6260009/139490368-9508e2e2-b98d-49f6-9a64-0bca0394f2f3.png">

**Comment by pubudu91** [comment]
This is actually not fixed in 2201.0.0. When we take the symbol of `rsa` and take its type descriptor, it returns a singleton type.
